### PR TITLE
Move Redux modules and entry point to `store/` directory

### PR DESCRIPTION
**Pull request by robertknight**
As per the plan outlined in #687, move the Redux store to a directory named `store/` and separate out the individual modules (in `store/modules/`) from utility code and middleware (in `store/`).

To keep the diff smaller, I haven't renamed references to "annotationUI" everywhere in this PR and will do that in a follow-up PR.

### CODE REVIEW

**Comment by hmstepanek** [comment]
Otherwise LGTM. Nice to see this change happening-it will be so much more organized and the naming makes more sense!

**Comment by robertknight** [comment]
> Otherwise LGTM. Nice to see this change happening-it will be so much more organized and the naming makes more sense!

Thanks for taking a look @hmstepanek. You said "otherwise" but I don't see any other comments. Did you have another comment that went missing or is that a typo?

**Comment by hmstepanek** [comment]
I made some comments on the code.

File: src/sidebar/index.js [hunk 1]
```diff
@@ -209,6 +208,10 @@ module.exports = angular.module('h', [
   .service('unicode', require('./services/unicode'))
   .service('viewFilter', require('./services/view-filter'))
 
+  // Redux store
+  .service('annotationUI', require('./store'))
```
[the remark this answers is not in the document]
Comment (robertknight) [reply src/sidebar/index.js]: Yes, in https://github.com/hypothesis/client/pull/695

File: src/sidebar/store/index.js [hunk 2]
```diff
@@ -72,8 +72,11 @@ function angularDigestMiddleware($rootScope) {
   };
 }
 
+/**
+ * Create the Redux store for the application.
+ */
 // @ngInject
-module.exports = function ($rootScope, settings) {
+function store($rootScope, settings) {
```
[the remark this answers is not in the document]
Comment (robertknight) [reply src/sidebar/store/index.js]: These are _almost_ equivalent, but the name is accessible via `f.name` and available in some debug contexts. I'm making this change mostly for consistency.

**Pull request merged**